### PR TITLE
add find_single_number

### PR DESCRIPTION
**Pull request by literalEval**
#### Description of Change

Added the question "Find non repeating number in an array of repeating numbers"

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Review by mishraabhinn — CHANGES_REQUESTED** [review]
Nice work @literalEval. Thanks for contributing.  The pr code LGTM! Can we please rethink the name of the file? After reading the name of file I can not understand the purpose.

**Comment by literalEval** [comment]
@mishraabhinn can you please suggest a better name ? 😅

**Comment by literalEval** [comment]
Hey @mishraabhinn can you please merge this PR ?

File: bit_manipulation/find_non_repeating_number.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+ * in an array of repeating integers. [Single
+ * Number](https://leetcode.com/problems/single-number/)
+ */
+namespace find_single_number {
+/**
+ * @brief The main function implements find single number
+ * @param nums vector of integers
+ * @returns returns the integer that occurs only once
+ */
+int find_single_number(std::vector<int>& nums) {
```
Comment (Panquesito7) [inline bit_manipulation/find_non_repeating_number.cpp:39]: You're not modifying `nums`, but you're accessing it. Using `const` will make the program faster and better.
```suggestion
int find_single_number(const std::vector<int>& nums) {
```

File: bit_manipulation/find_non_repeating_number.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+#include <iostream>  /// for IO operations
+#include <vector>    /// storing the numbers
+
+/**
+ * @namespace bit_manipulation
+ * @brief Bit manipulation algorithms
+ */
+namespace bit_manipulation {
+/**
+ * @namespace setKthBit
```
Comment (Panquesito7) [inline bit_manipulation/find_non_repeating_number.cpp:28]:
```suggestion
 * @namespace find_single_number
```

File: bit_manipulation/find_non_repeating_number.cpp (hunk 1 above)
Comment (Panquesito7) [inline bit_manipulation/find_non_repeating_number.cpp:39]: Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let us know if you need any help here or on our [Discord server](https://discord.gg/c7MnfGFGa6).** 🙂

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Great work! 🚀

**Comment by literalEval** [comment]
@Panquesito7  please review again

File: .vscode/settings.json [hunk 3]
```diff
@@ -1,64 +0,0 @@
-{
```
Comment (Panquesito7) [inline .vscode/settings.json:1]: Please add this file back as it originally was.

**Review by mishraabhinn — APPROVED** [review]
Approved. Nice work @literalEval. Please join discord and gitter for staying more connected.

**Comment by literalEval** [comment]
Done. Please merge this PR now. It has been hanging for a while now.

**Comment by mishraabhinn** [comment]
Waiting for another approval.

File: .vscode/settings.json [hunk 4]
```diff
@@ -1,64 +1,65 @@
...
+        "new": "cpp",
+        "ostream": "cpp",
+        "sstream": "cpp",
+        "stdexcept": "cpp",
+        "streambuf": "cpp",
+        "typeinfo": "cpp",
+        "valarray": "cpp"
+    },
+    "cmake.configureOnOpen": false
+}
```
Comment (Panquesito7) [inline .vscode/settings.json:65]:
```suggestion
}

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Comment by literalEval** [comment]
@Panquesito7 Done. Hope it is finally good now 🫠

**Review by mishraabhinn — DISMISSED** [review]

**Comment by literalEval** [comment]
@Panquesito7

**Review by Panquesito7 — APPROVED** [review]
Awesome work! Thank you so much. 🚀

**Pull request merged**